RebelAllianceBank/Classes/FileHandler.cs

```diff
@@ -0,0 +1,153 @@
+            if (File.Exists(filePath))
+            {
+                using (StreamReader sr = new StreamReader(filePath))
+                {
+                    string read;
+                    while ((read = sr.ReadLine()) != null)
+                    {
+                        string[] row = read.Split('-');
+                        IUser user = StoredUser(row);
+                        if (user != null)
+                        {
+                            savedList.Add(user);
+                        }
+                    }
+                }
+            }
+            return savedList;
+        }
+        public IUser StoredUser(string[] row)
+        {
+            switch (row[3])
+            {
+                case "true":
+                    return new Admin
+                    {
+                        ID = Convert.ToInt16(row[0]),
+                        Username = row[1],
+                        Password = row[2],
+                        IsAdmin = bool.Parse(row[3])
+                    };
+                case "false":
+                    return new Customer
+                    {
+                        ID = Convert.ToInt16(row[0]),
+                        Username = row[1],
+                        Password = row[2],
+                        IsAdmin = bool.Parse(row[3])
+                    };
+                default:
+                    return null;
+            }
+        }
+
+        public IUser WriteUserToFile(int id, string un, string pw, bool isAdmin)
+        {
+            string filePath = "..\\..\\..\\users.txt";
+            if (File.Exists(filePath))
+            {
+                using (StreamWriter sw = new StreamWriter(filePath, append: true))
+                {
+                    string write = $"{id}-{un}-{pw}-{isAdmin}";
+                    sw.WriteLine(write, Environment.NewLine);
+                }
+            }
+            IUser user = StoredUser(new string[] { id.ToString(), un, pw, isAdmin.ToString().ToLower() });
+            if (user != null)
+            {
+                return user;
+            }
+            return null;
+        }
+        public List<IBankAccount> ReadAccountsFromFile()
+        {
+            List<IBankAccount> savedList = new List<IBankAccount>();
+            string filePath = "..\\..\\..\\accounts.txt";
+            if (File.Exists(filePath))
+            {
+                using (StreamReader sr = new StreamReader(filePath))
+                {
+                    string read;
+                    while ((read = sr.ReadLine()) != null)
+                    {
+                        string[] row = read.Split('-');
+                        IBankAccount account = StoredBankAccount(row);
+                        if (account != null)
+                        {
+                            savedList.Add(account);
+                        }
+                    }
+                }
+            }
+            return savedList;
+        }
```

Comment:
This method can be simplify to a more general method same as on row 12.

Reply:
Merged methods to generic one